### PR TITLE
Upgrade CPU func tests to 2.o

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Upgrade AddConvertToReorderTest to 2.0
 - Upgrade AlignMatMulInputRanksTest to 2.0
 - Upgrade GatherAddAvgpool to 2.0
 - Upgrade InputNoReorderEltwiseBF16 to 2.0

### Tickets:
 - 125924

### CODE REVIEW

**Review by vurusovs — CHANGES_REQUESTED** [review]

File: src/common/transformations/src/transformations/convert_precision.cpp [hunk 1]
```diff
@@ -1267,6 +1267,8 @@ bool fuse_type_to_constant(const std::shared_ptr<ov::Node>& node,
             new_const = change_constant_precision<ov::element::Type_t::boolean, ov::element::Type_t::u8>(constant);
         } else if (from == ov::element::boolean && to == ov::element::i32) {
             new_const = change_constant_precision<ov::element::Type_t::boolean, ov::element::Type_t::i32>(constant);
+        } else if (from == ov::element::i8 && to == ov::element::i64) {
+            new_const = change_constant_precision<ov::element::Type_t::i8, ov::element::Type_t::i64>(constant);
```
Comment (riverlijunjie) [inline src/common/transformations/src/transformations/convert_precision.cpp:1271]: @vurusovs how do you think we change transformation to solve issue of ngraph::pass::ConvertPrecision?

Comment (vurusovs) [reply src/common/transformations/src/transformations/convert_precision.cpp:1271]: For me it looks great that we found such limitation. @itikhono please, take a look

**Review by vurusovs — APPROVED** [review]

File: src/common/transformations/src/transformations/convert_precision.cpp (hunk 1 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/convert_precision.cpp:1271]: this is a valid fix for this transformation.

But I'm not 100% sure, why we skipped some <from, to> pairs here.
E.g I don't see int8 to int32, uint8 to in64
Maybe the initial idea was not to have many template instantiations and add them on request.

Comment (riverlijunjie) [reply src/common/transformations/src/transformations/convert_precision.cpp:1271]: There are other similar issues, for example: https://github.com/openvinotoolkit/openvino/pull/21358#discussion_r1409020311

It convert i8/u8 to f32, also encounter the similar issue, should we handle it by the similar method? @itikhono

Comment (itikhono) [reply src/common/transformations/src/transformations/convert_precision.cpp:1271]: Are these only test relating issues or do we have some real models which require such changes?

**Review by ilya-lavrenov — APPROVED** [review]

File: src/common/transformations/src/transformations/convert_precision.cpp (hunk 1 above)
Comment (riverlijunjie) [reply src/common/transformations/src/transformations/convert_precision.cpp:1271]: I only found such issue in test cases, not sure whether there are any real models have such case.

**Pull request merged**